src/oldreader_api.cpp

```diff
@@ -66,24 +66,26 @@ std::string oldreader_api::retrieve_auth() {
 	}
 
 	std::string pass = cfg->get_configvalue("oldreader-password");
-    std::string pass_file = cfg->get_configvalue("oldreader-passwordfile");
-    std::string pass_eval = cfg->get_configvalue("oldreader-passwordeval");
+	std::string pass_file = cfg->get_configvalue("oldreader-passwordfile");
+	std::string pass_eval = cfg->get_configvalue("oldreader-passwordeval");
 	if ( pass == "" ) {
-        if ( pass_file != "" ) {
-            pass = read_password(pass_file.c_str());
-        } else if ( pass_eval != "") {
-            pass = eval_password(pass_eval.c_str());
-        }
+		if ( pass_file != "" ) {
+			pass = read_password(pass_file);
+			if ( pass == "" ) return "";
+		} else if ( pass_eval != "") {
+			pass = eval_password(pass_eval);
+			if ( pass == "" ) return "";
```

Comment:
> I can't think of some case in which it makes sense to have a prompt if the password file doesn't exist

But why, it makes perfect sense: if you don't set neither `password` nor `passwordfile`, you'll be prompted for a password every time. That's a nice way not to store the password anywhere.

> It's not very likely but it could break someones usage

I thought of that, but it looks like the only scenario that'll be affected is user not setting anything in their config and then pressing Enter in the prompt to use the empty password, which doesn't sound like a real-world use-case at all. Even if someone is testing things locally, they can always set one-char password to work around this.

And I'm not very haughty—I'll roll it back if users start complaining a lot :)